beacon_node/beacon_processor/src/lib.rs

```diff
@@ -1467,16 +1093,15 @@ impl<T: BeaconChainTypes> BeaconProcessor<T> {
                             Work::GossipLightClientOptimisticUpdate { .. } => {
                                 optimistic_update_queue.push(work, work_id, &self.log)
                             }
-                            Work::RpcBlock { .. } => rpc_block_queue.push(work, work_id, &self.log),
-                            Work::ChainSegment { ref process_id, .. } => match process_id {
-                                ChainSegmentProcessId::RangeBatchId { .. }
-                                | ChainSegmentProcessId::ParentLookup { .. } => {
-                                    chain_segment_queue.push(work, work_id, &self.log)
-                                }
-                                ChainSegmentProcessId::BackSyncBatchId { .. } => {
-                                    backfill_chain_segment.push(work, work_id, &self.log)
-                                }
-                            },
+                            Work::RpcBlock { .. } | Work::IgnoredRpcBlock { .. } => {
+                                rpc_block_queue.push(work, work_id, &self.log)
+                            }
+                            Work::ChainSegment { .. } => {
+                                chain_segment_queue.push(work, work_id, &self.log)
+                            }
+                            Work::ChainSegmentBackfill { .. } => {
+                                backfill_chain_segment.push(work, work_id, &self.log)
+                            }
```

Comment:
Previously we had a single `Work::ChainSegment` variant and we looked at the `process_id` to see if it's a backfill batch or not.

I've removed the need to look at `process_id` by creating two distinct variants; `Work::ChainSegment` and `Work::ChainSegmentBackfill`. I think it's cleaner this way, especially since we need to pass around different a `process_fn` depending on a forward/backfill segment.